Documentation/gettingstarted/k8s-install-openshift-okd.rst

```diff
@@ -118,54 +118,40 @@ Next, generate OpenShift manifests:
 
    openshift-install create manifests --dir "${CLUSTER_NAME}"
 
-Now, define ``cilium`` namespace:
+Next, obtain Cilium manifest from ``cilium/cilium-olm`` repository and copy to ``${CLUSTER_NAME}/manifests``:
 
-.. code:: bash
+.. parsed-literal::
 
-   cat << EOF > "${CLUSTER_NAME}/manifests/cluster-network-03-cilium-namespace.yaml"
-   apiVersion: v1
-   kind: Namespace
-   metadata:
-     name: cilium
-     annotations:
-       # node selector is required to make cilium-operator run on control plane nodes
-       openshift.io/node-selector: ""
-     labels:
-       name: cilium
-       # run level sets priority for Cilium to be deployed prior to other components
-       openshift.io/run-level: "0"
-       # enable cluster logging for Cilium namespace
-       openshift.io/cluster-logging: "true"
-       # enable cluster monitoring for Cilium namespace
-       openshift.io/cluster-monitoring: "true"
-   EOF
+   cilium_olm_rev=master
 
-.. include:: k8s-install-download-release.rst
+   curl --silent --location --fail --show-error "https://github.com/cilium/cilium-olm/archive/${cilium_olm_rev}.tar.gz" --output /tmp/cilium-olm.tgz
+   tar -C /tmp -xf /tmp/cilium-olm.tgz
 
-Next, render Cilium manifest:
+   cp /tmp/cilium-olm-${cilium_olm_rev}/manifests/cilium.v\ |release|\ /* "${CLUSTER_NAME}/manifests"
 
-.. parsed-literal::
+   rm -rf -- /tmp/cilium-olm.tgz "/tmp/cilium-olm-${cilium_olm_rev}"
 
-   helm template |CHART_RELEASE|  \\
-      --namespace cilium \\
-      --set ipam.mode=cluster-pool \\
-      --set cni.binPath=/var/lib/cni/bin \\
-      --set cni.confPath=/var/run/multus/cni/net.d \\
-      --set ipam.operator.clusterPoolIPv4PodCIDR=10.128.0.0/14 \\
-      --set ipam.operator.clusterPoolIPv4MaskSize=23 \\
-      --set nativeRoutingCIDR=10.128.0.0/14 \\
-      --set bpf.masquerade=false \\
-      --set endpointRoutes.enabled=true \\
-      --output-dir "${OLDPWD}"
-   cd "${OLDPWD}"
-
-Copy Cilium manifest to ``${CLUSTER_NAME}/manifests``:
+.. note::
 
-.. code:: bash
+   At this stage manifest directory contains all that is needed to install Cilium.
+   To get a list of the Cilium manifests, run ``ls ${CLUSTER_NAME}/manifests/cluster-network-*-cilium-*``.
+
+.. note::
+
+   If you wish to set any custom Helm values, you can do it by editing ``${CLUSTER_NAME}/manifests/cluster-network-07-cilium-ciliumconfig.yaml``.
+
+   It's also possible to update Helm values once the cluster is running by changing ``CiliumConfig`` object,
+   e.g. with  ``kubectl edit ciliumconfig -n cilium cilium``. You may need to restart Cilium agent pods for
+   certain options to take effect.
+
+.. note::
```

Comment:
Do we need to call these out in indented `note::` sections? Too many callouts will force users to ignore everything else. In general I would prefer to use these sparingly, and just write everything that the user may need to know directly in regular text (unless there's some exceptional circumstance that we really must highlight to users).

Reply:
I do generally agree, but some of the other callouts are really trying to address some nuances of different modes of installation, like cloud provider auths quirks that have little to do with Cilium, yet are gaps in OpenShift installer UX that I felt we should share with the users. This one in particular is kind of in the same territory, albeit a little more connected to Cilium, but it is mostly optional. Generally, I am quite sure that skipping most of the callouts will be fine for a lucky user who have installed OpenShift before.
We can review these more closely, but as I said in the other comment, the world of OpenShift torn with UX hindrances, and I don't think it's worse fixing. There are aspects worth addressing, but probably best to have separate discussions. Namely, we need to look at how to fix the issue with ports and drop hacky scripts that this guide currently asks user to run (exposing a flag to set VXLAN port would help there), and moving to our own operator is another aspect that can be addressed in 1.11.